src/App.jsx

```diff
@@ -1,31 +1,56 @@
 import './App.scss';
-import { MoviesList } from './components/MoviesList';
+import { useState } from 'react';
+
 import moviesFromServer from './api/movies.json';
+import { MoviesList } from './components/MoviesList';
+
+function getPreparedMoviesList(movies, { query }) {
+  let preparedMoviesList = [...movies];
+
+  const queryToLower = query.toLowerCase().trim();
+
+  if (query) {
+    preparedMoviesList = preparedMoviesList.filter(
+      movie =>
+        movie.title.toLowerCase().includes(queryToLower) ||
+        movie.description.toLowerCase().includes(queryToLower),
+    );
+  }
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+  return preparedMoviesList;
+}
+
+export const App = () => {
+  const [query, setQuery] = useState('');
+  const visibleMoviesList = getPreparedMoviesList(moviesFromServer, { query });
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={event => {
+                  setQuery(event.currentTarget.value);
+                }}
```

Comment:
Тут є два варіанта як можна покращити цей кусочок:
1) Винести оброку івента в окремий хендлер, там діставати таргет і викликати сеттер - це структурно гарний підхід, тому що якщо ми захочемо розширити цю функцію ми її будемо дописувати, а не переписувати велику частину
2)
```suggestion
                onChange={({ currentTarget }) => setQuery(currentTarget.value)}
```